### PR TITLE
sgt-back-end - Yuliia P.

### PR DESCRIPTION
**Pull request by yuliia-p**
Screen Shot 2022-08-25 at 6 09 31 PM
Screen Shot 2022-08-25 at 6 09 46 PM
Screen Shot 2022-08-25 at 6 11 18 PM
Screen Shot 2022-08-25 at 6 11 27 PM

### CODE REVIEW

File: sgt-back-end/index.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+      arrayOfObj.push(arrayOfObj);
+      res.status(200).json(grades);
+    }).catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+
+});
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js:68]: What's `arrayOfObj` for? It doesn't seem like it's being used for anything.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Comment by yuliia-p** [comment]
Screen Shot 2022-08-26 at 10 27 58 AM
Screen Shot 2022-08-26 at 10 29 18 AM
Screen Shot 2022-08-26 at 11 02 37 AM

File: sgt-back-end/index.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      .then(result => {
+        res.sendStatus(204);
+      }).catch(err => {
+        console.error(err);
+        res.status(500).json({
+          error: 'An unexpected error occurred.'
+        });
+      });
+  }
+});
```
[the remark this answers is not in the document]
Comment (yuliia-p) [reply sgt-back-end/index.js]: Screen Shot 2022-08-26 at 11 04 06 AM

File: sgt-back-end/index.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+app.listen(3000, () => {
+  // eslint-disable-next-line no-console
+  // console.log('Express server listening on port 3000');
```
[the remark this answers is not in the document]
Comment (yuliia-p) [reply sgt-back-end/index.js]: Screen Shot 2022-08-26 at 11 04 34 AM

File: sgt-back-end/index.js [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+        error: 'An unexpected error occurred.'
+      });
+    });
+
+});
+
+app.post('/api/grades/', (req, res) => {
+  const newGradeObj = req.body;
+  if (!newGradeObj.name || !newGradeObj.course || !newGradeObj.score) {
+    return res.status(400).json({ error: 'missing name, course, or score' });
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js:71]: A `score` of `0` should not trigger a `400 Bad Request`.

File: sgt-back-end/index.js [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    });
+
+});
+
+app.post('/api/grades/', (req, res) => {
+  const newGradeObj = req.body;
+  if (!newGradeObj.name || !newGradeObj.course || !newGradeObj.score) {
+    res.status(400).json({ error: 'missing name, course, or score' });
+  } else if ((!Number.isInteger(newGradeObj.score) || newGradeObj.score > 100)) {
+    res.status(400).json({ error: 'invalid score' });
```
[the remark this answers is not in the document]
Comment (yuliia-p) [reply sgt-back-end/index.js]: Screen Shot 2022-08-26 at 11 05 47 AM

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: sgt-back-end/index.js (hunk 4 above)
Comment (yuliia-p) [reply sgt-back-end/index.js:71]: Screen Shot 2022-08-26 at 11 17 27 AM

Comment (yuliia-p) [reply sgt-back-end/index.js]: Screen Shot 2022-08-26 at 11 18 37 AM

File: sgt-back-end/index.js [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    });
+
+});
+
+app.post('/api/grades/', (req, res) => {
+  const newGradeObj = req.body;
+  if (!newGradeObj.name || !newGradeObj.course) {
+    return res.status(400).json({ error: 'missing name, course, or score' });
+  } else if ((!Number.isInteger(newGradeObj.score) || newGradeObj.score > 100 || newGradeObj.score < -1)) {
+    return res.status(400).json({ error: 'invalid score' });
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js:73]: A `score` equal to `-1` should result in a `400 Bad Request`

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: sgt-back-end/index.js (hunk 6 above)
Comment (yuliia-p) [reply sgt-back-end/index.js]: Screen Shot 2022-08-26 at 11 28 09 AM
Screen Shot 2022-08-26 at 11 28 55 AM

File: sgt-back-end/index.js [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+      });
+    });
+});
+
+app.put('/api/grades/:gradeId', (req, res) => {
+  const idToUpdate = Number(req.params.gradeId);
+  if (!Number.isInteger(idToUpdate) || idToUpdate <= 0) {
+    return res.status(400).json({ error: 'id must be a positive integer' });
+  } else if (!req.body.name || !req.body.course || !req.body.score) {
+    return res.status(400).json({ error: 'missing name, course, or score' });
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js:97]: A `score` of `0` should not trigger a `400 Bad Request`.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: sgt-back-end/index.js (hunk 7 above)
Comment (yuliia-p) [reply sgt-back-end/index.js]: Screen Shot 2022-08-26 at 12 21 26 PM

Comment (yuliia-p) [reply sgt-back-end/index.js]: Screen Shot 2022-08-26 at 12 23 04 PM

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**